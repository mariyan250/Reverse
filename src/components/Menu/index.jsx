// React and Style
import React from 'react';
import styles from './index.module.scss';

// Redux
import { connect } from 'react-redux';
import { removeUser } from 'store/actions/user';

// Translation
import { useTranslation } from 'react-i18next';

// Router
import { NavLink, useHistory } from 'react-router-dom';

function Menu({ links, removeUser, user }) {
  const { t } = useTranslation();
  const history = useHistory();

  const handleLogout = () => {
    removeUser();
    history.push('/');
  };

  return (
    <ul className={`${styles.menu} app-bg-black-transparent`}>
      {links.map((link, i) => (
        <li key={i}>
          <NavLink
            to={link.to}
            className="d-flex w-100 justify-content-center align-items-center app-text-white app-bg-black-transparent py-4"
          >
            {t(link.text)}
          </NavLink>
        </li>
      ))}

      {user && (
        <li>
          <button
            onClick={handleLogout}
            data-testid="logout"
            aria-label="logout"
            className="d-flex justify-content-center align-items-center text-center w-100 app-text-white app-bg-black-transparent py-4 border-0"
          >
            {t('header.logout_btn')}
            <i className="fas fa-sign-out-alt ml-3" />
          </button>
        </li>
      )}
    </ul>
  );
}

const mapStateToProps = (state) => ({
  user: state.user,
});

export default connect(mapStateToProps, { removeUser })(Menu);
