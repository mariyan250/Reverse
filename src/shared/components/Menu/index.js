// React and Style
import React from 'react';
import './index.scss';

// Router
import { NavLink } from 'react-router-dom';

function Menu(props) {
  return (
    <section {...props}>
      <article className="menu">
        <ul className="d-flex flex-column w-100 h-100">
          <li className="d-flex">
            <NavLink
              to="/dashboard"
              className="menu-link d-flex justify-content-center align-items-center text-decoration-none"
              activeClassName="link-active"
            >
              <i className="fas fa-home mr-3" />
              Home
            </NavLink>
          </li>

          <li className="d-flex">
            <NavLink
              to="/login"
              className="menu-link d-flex justify-content-center align-items-center text-decoration-none"
              activeClassName="link-active"
            >
              <i className="fas fa-folder mr-3" />
              My Items
            </NavLink>
          </li>

          <li className="d-flex">
            <NavLink
              to="/register"
              className="menu-link d-flex justify-content-center align-items-center text-decoration-none"
              activeClassName="link-active"
            >
              <i className="fas fa-envelope mr-3" />
              Messages
            </NavLink>
          </li>

          <li className="d-flex">
            <NavLink
              to="/register"
              className="menu-link d-flex justify-content-center align-items-center text-decoration-none"
              activeClassName="link-active"
            >
              <i className="fas fa-cog mr-3" />
              Settings
            </NavLink>
          </li>
        </ul>
      </article>
    </section>
  );
}

export default Menu;
