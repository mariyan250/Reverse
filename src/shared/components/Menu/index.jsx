// React and Style
import React from 'react';
import styles from './index.module.scss';

// Translation
import { useTranslation } from 'react-i18next';

// Router
import { NavLink } from 'react-router-dom';

function Menu(props) {
  const { t } = useTranslation();

  return (
    <div {...props}>
      <div className={`${styles.menu} app-bg-black-transparent`}>
        <ul>
          {props.links &&
            props.links.map((link, i) => (
              <li className={`d-flex`} key={i}>
                <NavLink
                  to={link.to}
                  className={`d-flex w-100 justify-content-center align-items-center app-text-white app-bg-black-transparent py-4`}
                >
                  <i className={`${link.iconName} mr-3`} />
                  {t(link.text)}
                </NavLink>
              </li>
            ))}
        </ul>
      </div>
    </div>
  );
}

export default Menu;
