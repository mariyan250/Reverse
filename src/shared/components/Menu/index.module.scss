@import '../../../index.scss';

.menu {
  a {
    font-size: to-rem(16px);

    i {
      font-size: to-rem(22px);
    }
  }
}
