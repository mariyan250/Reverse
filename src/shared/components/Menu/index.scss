@import '../../../index.scss';

.menu {
  background: $background-black-transparent;
  height: 18rem;

  ul {
    li {
      flex: 1;

      .menu-link {
        flex: 1;
        color: $text-white;
        font-size: 16px;

        i {
          font-size: 22px;
        }
      }

      .menu-link:hover,
      .link-active {
        background: $background-white;
        color: $black;
      }
    }
  }
}
