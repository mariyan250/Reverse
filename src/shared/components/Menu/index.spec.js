import React from 'react';
import { render } from 'tests/test-utils';
import Menu from './index';

describe('Menu', () => {
  it('should render the component', () => {
    const { container } = render(<Menu links={[]} />);
    expect(container).toMatchSnapshot();
  });
});
