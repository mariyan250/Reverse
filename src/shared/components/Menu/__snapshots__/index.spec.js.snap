// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render the component 1`] = `
<div>
  <div
    links=""
  >
    <div
      class="menu app-bg-black-transparent"
    >
      <ul />
    </div>
  </div>
</div>
`;
